core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -2856,6 +2856,32 @@ LogicalAggregate(group=[{}], EXPR$0=[MIN($1) FILTER $3], EXPR$1=[COUNT($0) FILTE
     LogicalAggregate(group=[{1, 2}], groups=[[{1, 2}, {}]], EXPR$0=[SUM($0)], $g=[GROUPING($1, $2)])
       LogicalProject(COMM=[$6], SAL=[$5], $f2=[>($5, 1000)])
         LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+]]>
+    </Resource>
+  </TestCase>
+  <TestCase name="testDistinctWithFilterWithoutGroupByUsingJoin">
+    <Resource name="sql">
+      <![CDATA[SELECT SUM(comm), COUNT(DISTINCT sal) FILTER (WHERE sal > 1000)
+FROM emp]]>
+    </Resource>
+    <Resource name="planBefore">
+      <![CDATA[
+LogicalAggregate(group=[{}], EXPR$0=[SUM($0)], EXPR$1=[COUNT(DISTINCT $1) FILTER $2])
+  LogicalProject(COMM=[$6], SAL=[$5], $f2=[>($5, 1000)])
+    LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+]]>
+    </Resource>
+    <Resource name="planAfter">
+      <![CDATA[
+LogicalJoin(condition=[true], joinType=[inner])
```

Comment:
frankly, this plan looks wrong to me.
From my reading of the spec the filter should not be applied to the sum.

Reply:
The filter is not being applied to the sum. There is an additional `project` for the count aggregator that looks at the result of the filter condition and projects another column that gets counted. If the filter were being applied to the sum, I would imagine this `case` project being on the left side of the join, too. 

I haven't looked at these plans before, but please let me know if I am misinterpreting them. I have also verified this fix in `apache/druid` and I could see the native druid plan adding the filter only to the count.

Reply:
My mistake.

Should we strive to address [CALCITE-6332] in this PR as well? I saw you commented on it.
As far as I can tell, a rule can check some preconditions and refuse to be applied if they are not satisfied. That would be one way to solve 6332.

As for this particular fix, don't know what happens if you have multiple aggregates each with a different condition, as the other above comment from @kgyrtkirk points out.

One solution would be to refuse to apply this rule when such combinations of filters exist.

I also sent a question to the dev list about the ability to test such rules using Quidem tests. That would be great, but I don't know if it's possible yet.

Reply:
That (refusing to let the JOIN flavour fire when query has grouping sets and distinct counts) should be done in a separate PR. I will try a test case for the scenario when multiple aggregates with different conditions are passed and see if Zoltan's suggestion fixes the issues.

Reply:
What is the status of this PR? @abhishekagarwal87 are you still working on this?

Reply:
It should be ready now.